File: src/types/portfolio.ts
export interface PortfolioStock {
    id: number
    userId: number
    symbol: string
    quantity: number
    purchasePrice: number
    purchaseDate: string
    notes?: string
  }

  export interface UserOwnedStock {
    id: number
    userId: number
    symbol: string
    quantity: number
    averagePurchasePrice: number
    totalCost: number
    currentValue: number
    gainLoss: number
    gainLossPercent: number
    purchaseDate: string
    notes?: string
  }
  
  

File: api/Models/Portfolio.cs
namespace api.Models;

public class Portfolio
{
    public int Id { get; set; }
    public required int UserId { get; set; }
    public required string Symbol { get; set; }
    public required decimal Quantity { get; set; }
    public required decimal PurchasePrice { get; set; }
    public required DateTime PurchaseDate { get; set; }
    public string? Notes { get; set; }
    public User User { get; set; } = null!;
}
